### PR TITLE
Update French content for Access to dental insurance page (route 7)

### PR DESCRIPTION
**Pull request by shawn320**
### Description
Add French text for Access to dental insurance page (route 7)

### Checklist

- [ ] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [ ] I have checked that my code follows the project's coding style by running `npm run format:check`
- [ ] I have checked that my code contains no linting errors by running `npm run lint`
- [ ] I have checked that my code contains no type errors by running `npm run typecheck`
- [ ] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/public/locales/fr/apply.json [hunk 1]
```diff
@@ -157,34 +157,34 @@
...
-      "continue": "(Fr) Continue"
+      "back": "Retour",
+      "continue": "Continuer"
     },
     "detail": {
       "additional-info": {
-        "title": "(Fr) Additional information",
-        "eligible": "(Fr) Exception: You may still be eligible if you retired and:",
+        "title": "Renseignements supplémentaires",
+        "eligible": "Exception : vous êtes admissible si vous avez pris votre retraite et que :",
```
Comment (sebastien-comeau) [inline frontend/public/locales/fr/apply.json:177]: Add non-breaking space character `\u00a0` before colon in French.

File: frontend/public/locales/fr/apply.json
Comment (sebastien-comeau) [inline frontend/public/locales/fr/apply.json:1]: Replace all `’` curly apostrophes with straight appostrophe `'`.

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/public/locales/fr/apply.json
Comment (sebastien-comeau) [reply frontend/public/locales/fr/apply.json:1]: Not fixed

Comment (shawn320) [reply frontend/public/locales/fr/apply.json:1]: Which part? I replaced all `’` for dental-insurance

Comment (sebastien-comeau) [reply frontend/public/locales/fr/apply.json:1]: Oh, chrome doesn't know the different between them when searching. You fixed it. :)

Comment (shawn320) [reply frontend/public/locales/fr/apply.json:1]: Sorry, I think I misunderstood. They should be straight or curly?

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**